packages/x-charts/src/ChartsXAxis/ChartsXAxis.tsx

```diff
@@ -175,12 +173,41 @@ function ChartsXAxis(inProps: ChartsXAxisProps) {
     tickPlacement,
     tickLabelPlacement,
   });
+  const [measurements, setMeasurements] = React.useState(
+    new Map<number, { width: number; height: number }>(),
+  );
+
+  React.useLayoutEffect(
+    function measureTicks() {
+      if (needsMeasuring) {
+        setMeasurements(
+          new Map(
+            xTicks.map(function measureTick(_, index) {
+              const bbox = textMapRef.current.get(index)?.getBBox() ?? {
+                width: 0,
+                height: 0,
+              };
+
+              return [index, { width: bbox.width, height: bbox.height }] as const;
+            }),
+          ),
+        );
+        setNeedsMeasuring(false);
+      }
+    },
+    [needsMeasuring, xTicks],
+  );
 
-  const xTicksWithDimension = addLabelDimension(xTicks, {
+  if (!objectShallowCompare(prevTickLabelStyle, axisTickLabelProps.style)) {
```

Comment:
Need to shallow compare to avoid issues with object equality. Not sure if we have a preferred way of doing this in the charts package. I didn't find a shallow compare function in our package, so importing the one from data grid for now.

Reply:
> Need to shallow compare to avoid issues with object equality. Not sure if we have a preferred way of doing this in the charts package. I didn't find a shallow compare function in our package, so importing the one from data grid for now.

import from `x-internals`